editor/block-switcher/index.js

```diff
@@ -72,38 +47,21 @@ class BlockSwitcher extends Component {
 		}
 
 		return (
-			<div className="editor-block-switcher">
-				<IconButton
-					className="editor-block-switcher__toggle"
-					icon={ blockType.icon }
-					onClick={ this.toggleMenu }
-					aria-haspopup="true"
-					aria-expanded={ this.state.open }
-					label={ __( 'Change block type' ) }
-				>
-					<Dashicon icon="arrow-down" />
-				</IconButton>
-				{ this.state.open &&
-					<div
-						className="editor-block-switcher__menu"
-						role="menu"
-						tabIndex="0"
-						aria-label={ __( 'Block types' ) }
-					>
-						{ allowedBlocks.map( ( { name, title, icon } ) => (
-							<IconButton
-								key={ name }
-								onClick={ this.switchBlockType( name ) }
-								className="editor-block-switcher__menu-item"
-								icon={ icon }
-								role="menuitem"
-							>
-								{ title }
-							</IconButton>
-						) ) }
-					</div>
-				}
-			</div>
+			<Toolbar>
+				<li>
```

Comment:
Aside: Ideally a consumer of the `Toolbar` component should not care that it renders itself as a `ul`, and could pass children without the `li`. In the implementation of the Toolbar component, children can be achieved with `React.Children.map`:

```jsx
<ul>
	{ React.Children.map( ( child, i ) => (
		<li key={ i }>{ child }</li>
	) ) }
</ul>
```

_Mentioning as aside since it's not really relevant to the scope of your pull request._

Reply:
>however it would also remove some control from the caller (ie to add classes to some li elements).

To me, this is not control we want to allow, or at least not through exposing `li` directly. There are a few alternatives, such as a prop on `Toolbar`, a wrapper for each children like `ToolbarItem`, a wrapper of the item itself (apply class to a `div` parent of `DropdownMenu`), or more advanced examples like [function as child components](https://medium.com/merrickchristensen/function-as-child-components-5f3920a9ace9).

Ultimately, I don't think we know enough about the actual use cases to need to concern ourselves with this control just yet though.